### ISSUE TITLE
Output of `disjointBins()` is assigned the wrong names if input is not sorted

### ISSUE THREAD

**Issue by csoneson** [Bioconductor/IRanges#54]
Hi,

when calling `disjointBins()` on an unsorted `IRanges` object, I think the values in the output vector are indeed parallel to the input object, but the names don't match. For example:

```
ir <- IRanges(
    start = c(101, 27, 10123, 10245, 32138, 28, 27, 29539, 39010, 28, 10991, 35716),
    end = c(8342, 8022, 13608, 40902, 42373, 36713, 1007, 33072, 42114, 49971, 32495, 36417),
    names = paste0("R", 1:12)
)
djb <- disjointBins(ir)
ir[c("R2", "R10")]
djb[c("R2", "R10")]
djb
```

gives the following: 
```
## two overlapping ranges are still assigned to the same bin
> ir[c("R2", "R10")]
IRanges object with 2 ranges and 0 metadata columns:
          start       end     width
      <integer> <integer> <integer>
   R2        27      8022      7996
  R10        28     49971     49944
> djb[c("R2", "R10")]
 R2 R10
  2   2
> djb
 R7  R2  R6 R10  R1  R3  R4 R11  R8  R5 R12  R9
  5   2   1   2   6   3   1   1   1   4   5   1
```

The names of `djb` agree with the names of the input ranges sorted by start position. I think it happens here (bins ordering is reverted, name ordering is not): https://github.com/Bioconductor/IRanges/blob/2efdf2e5b992792a4a0ea97afb13e958e2c6304b/R/inter-range-methods.R#L509-L521

This happens for current release (2.40.0) and devel (2.41.1) versions.

**Comment by hpages** [comment]
Thanks @csoneson for the report.

Should be fixed in **IRanges** 2.40.1 (release) and 2.41.2 (devel). See commit 4eff83a94e30320f225dacdea9654d37b623e99f

If all goes well, both versions will become available via `BiocManager::install()` in the next 48 hours or so.

**Comment by csoneson** [comment]
Brilliant, thanks @hpages for the quick fix! 🙂
